haystack/dataclasses/document.py

```diff
@@ -22,27 +24,23 @@ class _BackwardCompatible(type):
 
     def __call__(cls, *args, **kwargs):
         """
-        Called before Document.__init__, will remap legacy fields to new ones.
+        Called before Document.__init__, handles legacy fields.
 
-        Also handles building a Document from a flattened dictionary.
-        Dataframe is not supported anymore.
+        Embedding was stored as NumPy arrays in 1.x, so we convert it to a list of floats.
+        Other legacy fields are removed.
         """
         ### Conversion from 1.x Document ###
         content = kwargs.get("content")
         if content and not isinstance(content, str):
             raise ValueError("The `content` field must be a string or None.")
 
-        # Not used anymore
-        if "content_type" in kwargs:
-            del kwargs["content_type"]
```

Comment:
Is there a reason to remove this? Without this if content_type is present in the dict then this would fail right?

Reply:
Okay I see this isn't needed for the tests to pass which explicitly checks this, but don't fully get why.

Reply:
this code block would do the same for all legacy fields, no?

```python
        # Remove legacy fields
        for field_name in LEGACY_FIELDS:
            kwargs.pop(field_name, None)
```